api/src/Microsoft.Azure.IIoT.Api/src/Publisher/Clients/MonitoredItemMessagePublisher.cs

```diff
@@ -39,25 +39,21 @@ public sealed class MonitoredItemMessagePublisher : ISubscriberMessageProcessor,
             foreach (var datapoint in message.Payload) {
                 var arguments = new object[] {
                      new MonitoredItemMessageApiModel() {
-                        Value = datapoint.Value.TypeId?.IsPrimitive == true ?
-                            datapoint.Value.Value : datapoint.Value.Value?.ToString(),
-                        TypeId = datapoint.Value.TypeId?.FullName,
```

Comment:
> TypeId  [](start = 23, length = 8)

is typeid not set anymore?

Reply:
yes, TypeId is removed since the OPC UA types are not known at this level anymore. We rely on .net types

Reply:
but the model still contains TypeId

---
In reply to: [397342767](https://github.com/Azure/Industrial-IoT/pull/327#discussion_r397342767) [](ancestors = 397342767)

Reply:
I did not change the apimodel because this is still consumed by the engineering tool, and I tried to limit the refactoring to what's essential. Eventually we will need to get rid of this class completely, but one thing at the time.

Reply:
Let's be careful - .net types should not be exposed in the API model.   Only OPC UA types.   How we deduce a .net type should be internal magic and not on the wire (for security and usability reasons).   Let's review everything with this in mind.

Reply:
@marcschier, totally agree, the TypeId is already removed from the Pub/Sub DataSet message - which will be the unique format through processors. The MonitoredItemMessageApiModel is still there only because it is consumed by the Engineering Tool. I propose I'll take the step of removing it from SignalR and EngTool in a separate PR.

Reply:
👍